src/onnx/parse_multinomial.cpp

```diff
@@ -46,38 +46,79 @@ struct parse_multinomial : op_parser<parse_multinomial>
             dtype = info.attributes.at("dtype").i();
         shape::type_t output_type = get_type(dtype);
 
+        size_t batch_size = 1;
+        if(contains(info.attributes, "batch_size"))
+            batch_size = info.attributes.at("batch_size").i();
+
         size_t sample_size = 1;
         if(contains(info.attributes, "sample_size"))
             sample_size = info.attributes.at("sample_size").i();
+        else
+            MIGRAPHX_THROW("PARSE_MULTINOMIAL: sample_size not given");
 
         // Subtract the per-batch maximum log-probability, making the per-batch max 0
         auto maxes =
             info.add_instruction(migraphx::make_op("reduce_max", {{"axes", {1}}}), args[0]);
-        auto mb_maxes = info.add_instruction(
-            migraphx::make_op("multibroadcast", {{"out_lens", args[0]->get_shape().lens()}}),
-            maxes);
-        auto cdf = info.add_instruction(migraphx::make_op("sub"), args[0], mb_maxes);
+        auto cdf = info.add_common_op("sub", args[0], maxes);
         // Take the element-wise exponent to get probabilities in the range (0, 1]
         cdf = info.add_instruction(migraphx::make_op("exp"), cdf);
         // Compute the cumulative density function
         cdf = info.add_instruction(
             migraphx::make_op("prefix_scan_sum", {{"axis", 1}, {"exclusive", false}}), cdf);
 
-        // Pre-compute random distribution
-        std::mt19937 gen(std::chrono::high_resolution_clock::now().time_since_epoch().count());
+        instruction_ref seed_input;
         if(contains(info.attributes, "seed"))
-            gen.seed(info.attributes.at("seed").f());
+        {
+            uint32_t seed = info.attributes.at("seed").i();
+            migraphx::shape s{migraphx::shape::uint32_type, {1}};
+            std::vector<uint32_t> data = {seed};
```

Comment:
ONNX spec says the `seed` attribute is a float. This will convert to a `uint32_t`. I'm not sure what's the intended behavior by the ONNX  spec. There will be a problem with the current `random_seed` op if we request it to create a floating point value since `std::random_device` only does integers. We probably want to check the accuracy of `multinomial` against ORT or something in a `verify_onnx.cpp` test.

Reply:
I think Paul kept asking me to make changes to the seed type for the op and in one version the op only supported one type, but I don't think there's any obstacle to changing it to a float here now.  I just want to make sure we agree on what we're trying to accomplish before I make the change.

Reply:
I would have to change some tests and Onnx test files to match the change.

Reply:
Changed the type of seed in parse_multinomial to float.  Should be transparent to the MIGraphX ops.